internal/services/apimanagement/api_management_policy_resource.go

```diff
@@ -105,10 +111,12 @@ func resourceApiManagementPolicyCreateUpdate(d *pluginsdk.ResourceData, meta int
 		return fmt.Errorf("Either `xml_content` or `xml_link` must be set")
 	}
 
-	if _, err := client.CreateOrUpdate(ctx, resourceGroup, serviceName, parameters, ""); err != nil {
+	policyResp, err := client.CreateOrUpdate(ctx, resourceGroup, serviceName, parameters, "")
+	if err != nil {
 		return fmt.Errorf("creating or updating Policy (Resource Group %q / API Management Service %q): %+v", resourceGroup, serviceName, err)
 	}
 
+	id := parse.NewPolicyID(apiMgmtId.SubscriptionId, apiMgmtId.ResourceGroup, apiMgmtId.ServiceName, utils.NormalizeNilableString(policyResp.Name))
```

Comment:
since we're defining this, should we be hard-coding this?

```suggestion
	id := parse.NewPolicyID(apiMgmtId.SubscriptionId, apiMgmtId.ResourceGroup, apiMgmtId.ServiceName, "policy")
```

Reply:
related: #15087 may conflict with this, since IIRC it moves this line upwards (to return it as the Requires Import error)